### PR TITLE
xxhash cleanups

### PR DESCRIPTION
**Pull request by sleeepyjack**
This PR is a follow-up on #310 and addresses review comments from @bdice.

### CODE REVIEW

File: include/cuco/detail/hash_functions/xxhash.cuh [hunk 1]
```diff
@@ -86,20 +86,20 @@ struct XXHash_32 {
   constexpr result_type __host__ __device__ operator()(Key const& key) const noexcept
   {
     // TODO do we need to add checks/hints for alignment?
```
Comment (sleeepyjack) [inline include/cuco/detail/hash_functions/xxhash.cuh:88]: Can someone comment on this? I'm not sure if the excessive `reinterpret_cast`ing leads to pitfalls.

File: include/cuco/detail/hash_functions/xxhash.cuh [hunk 2]
```diff
@@ -60,22 +60,22 @@ namespace cuco::detail {
...
-  static constexpr uint32_t prime1 = 0x9E3779B1U;
-  static constexpr uint32_t prime2 = 0x85EBCA77U;
-  static constexpr uint32_t prime3 = 0xC2B2AE3DU;
-  static constexpr uint32_t prime4 = 0x27D4EB2FU;
-  static constexpr uint32_t prime5 = 0x165667B1U;
+  static constexpr std::uint32_t prime1 = 0x9E3779B1u;
+  static constexpr std::uint32_t prime2 = 0x85EBCA77u;
+  static constexpr std::uint32_t prime3 = 0xC2B2AE3Du;
+  static constexpr std::uint32_t prime4 = 0x27D4EB2Fu;
+  static constexpr std::uint32_t prime5 = 0x165667B1u;
```
Comment (PointKernel) [inline include/cuco/detail/hash_functions/xxhash.cuh]: I think @bdice's idea was to use lowercase consistently, e.g. 
```suggestion
  static constexpr std::uint32_t prime1 = 0x9e3779b1u;
  static constexpr std::uint32_t prime2 = 0x85ebca77u;
  static constexpr std::uint32_t prime3 = 0xc2b2ae3du;
  static constexpr std::uint32_t prime4 = 0x27d4eb2fu;
  static constexpr std::uint32_t prime5 = 0x165667b1u;
```
Not a hard requirement though

File: include/cuco/detail/hash_functions/xxhash.cuh (hunk 1 above)
Comment (PointKernel) [reply include/cuco/detail/hash_functions/xxhash.cuh:88]: cc @jrhemstad

File: include/cuco/detail/hash_functions/xxhash.cuh [hunk 3]
```diff
@@ -314,13 +314,13 @@ struct XXHash_64 {
     // remaining data can be processed in 4-byte chunks
     if constexpr (((nbytes % 32) % 8) >= 4) {
       for (; offset <= nbytes - 4; offset += 4) {
-        h64 ^= (blocks4[offset / 4] & 0xFFFFFFFFULL) * prime1;
+        h64 ^= (blocks4[offset / 4] & 0xFFFFFFFFull) * prime1;
```
Comment (PointKernel) [inline include/cuco/detail/hash_functions/xxhash.cuh]:
```suggestion
        h64 ^= (blocks4[offset / 4] & 0xffffffffull) * prime1;
```

File: include/cuco/detail/hash_functions/xxhash.cuh (hunk 1 above)
Comment (PointKernel) [reply include/cuco/detail/hash_functions/xxhash.cuh:88]: Based on the offline discussion, downcasting to the char pointer shouldn't cause any problems and `blocks` (or the `uint32_t` pointer) won't be used if the key size is smaller than 4 bytes so the current implementation should be safe.

File: include/cuco/detail/hash_functions/xxhash.cuh (hunk 3 above)
Comment (sleeepyjack) [reply include/cuco/detail/hash_functions/xxhash.cuh]: fixed in [00e5b54](https://github.com/NVIDIA/cuCollections/pull/313/commits/00e5b54eb31aba90d396552c2552c3492c98fe0f)

File: include/cuco/detail/hash_functions/xxhash.cuh (hunk 2 above)
Comment (sleeepyjack) [reply include/cuco/detail/hash_functions/xxhash.cuh]: fixed in [00e5b54](https://github.com/NVIDIA/cuCollections/pull/313/commits/00e5b54eb31aba90d396552c2552c3492c98fe0f)

File: include/cuco/detail/hash_functions/xxhash.cuh (hunk 1 above)
Comment (sleeepyjack) [reply include/cuco/detail/hash_functions/xxhash.cuh:88]: [8c4f167](https://github.com/NVIDIA/cuCollections/pull/313/commits/8c4f1673f09a77ebbb12cc43ea4275410b37b0cb) sprinkles in some `[[maybe_unused]]` attributes for good measure.

**Review by PointKernel — APPROVED** [review]
LGTM

**Pull request merged**